tests/drivers/dac/dac_api/src/test_dac.c

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2020 Libre Solar Technologies GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*
+ * @addtogroup test_dac_basic_operations
+ * @{
+ * @defgroup t_dac_basic_basic_operations test_dac_write
+ * @brief TestPurpose: verify DAC driver handles single write
+ * @}
+ */
+
+#include <drivers/dac.h>
+#include <zephyr.h>
+#include <ztest.h>
+
+#if defined(CONFIG_BOARD_NUCLEO_L073RZ)
+
+#define DAC_DEVICE_NAME		DT_ALIAS_DAC1_LABEL
+#define DAC_RESOLUTION		12
+#define DAC_REFERENCE		DAC_REF_DEFAULT
+#define DAC_1ST_CHANNEL_ID	1
+
+#else
+#error "Unsupported board."
+#endif
+
+static const struct dac_channel_cfg dac_ch1_cfg = {
+	.channel_id  = DAC_1ST_CHANNEL_ID,
+	.reference = DAC_REFERENCE,
+	.buffered = false
+};
+
+struct device *get_dac_device(void)
+{
+	return device_get_binding(DAC_DEVICE_NAME);
+}
+
+static struct device *init_dac(void)
+{
+	int ret;
+	struct device *dac_dev = device_get_binding(DAC_DEVICE_NAME);
+
+	zassert_not_null(dac_dev, "Cannot get DAC device");
+
+	ret = dac_channel_setup(dac_dev, &dac_ch1_cfg);
+	zassert_equal(ret, 0,
+		"Setting up of the first channel failed with code %d", ret);
+
+	return dac_dev;
+}
+
+/*
+ * test_dac_write_value
+ */
+static int test_task_write_value(void)
+{
+	int ret;
+
+	struct device *dac_dev = init_dac();
+
+	if (!dac_dev) {
+		return TC_FAIL;
+	}
+
+	ret = dac_write_value(dac_dev, DAC_1ST_CHANNEL_ID, 2048, 12);
```

Comment:
`12` needs to be `DAC_RESOLUTION`. It would be ideal if the `2048` was half of the full scale for any resolution (here it is hardcoded).